fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
 
         final Collection<IllegalArgumentException> errors = checkInvalidPredicates(request);
 
+        request.getPrefixMapping().getNsPrefixMap().forEach(
+                       (k,v) -> { try {
+                                   LOGGER.info("451: prefix mapping is key -> value: " + k + "->" + v + "\n");
+                  if ( Arrays.asList(getSession().getWorkspace().getNamespaceRegistry().getPrefixes()).contains(k)
+                     && !v.equals(getSession().getWorkspace().getNamespaceRegistry().getURI(k))) {
```

Comment:
Likewise, you could probably retrieve the namespace-registry before this stream, and reuse it here.